api/src/main/kotlin/io/spine/protodata/ProtoDeclaration.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package io.spine.protodata
+
+import com.google.protobuf.Message
+import io.spine.annotation.GeneratedMixin
+import io.spine.annotation.Internal
+
+/**
+ * A high-level Protobuf declaration, such as a message, an enum, or a service.
+ */
+@Internal
+@GeneratedMixin
+public interface ProtoDeclaration : Message {
+
+    /**
+     * Obtains the name of this Protobuf declaration.
+     */
+    public val name: ProtoDeclarationName
+
+    /**
+     * Obtains the type URL of the type.
+     *
+     * A type URL contains the type URL prefix and the qualified name of the type separated by
+     * the slash (`/`) symbol. See the docs of `google.protobuf.Any.type_url` for more info.
+     */
+    public fun typeUrl(): String {
```

Comment:
Why don't we have it as another property rather than a function? Just curious, how you see it.

One reason I can think of is to have shorter names in Java code. But we aim to Kotlin anyway, and we always can have `@JvmName` defined, should we want to.